`test_storage_rabbitmq` is broken

It's been broken for 2 months already: https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZEYXkoY2hlY2tfc3RhcnRfdGltZSkgYXMgZCwKY291bnQoKSwgIGdyb3VwVW5pcUFycmF5KHB1bGxfcmVxdWVzdF9udW1iZXIpLCAgYW55KHJlcG9ydF91cmwpCmZyb20gY2hlY2tzIHdoZXJlICcyMDIyLTA2LTAxJyA8PSBjaGVja19zdGFydF90aW1lIGFuZCB0ZXN0X25hbWUgbGlrZSAnJXRlc3Rfc3RvcmFnZV9yYWJiaXRtcSUnIGFuZCB0ZXN0X3N0YXR1cyBpbiAoJ0ZBSUwnLCAnRkxBS1knKSBncm91cCBieSBkIG9yZGVyIGJ5IGQgZGVzYw==

Similar as the latest mentions (https://s3.amazonaws.com/clickhouse-test-reports/69780/6ba4f61df86c8edc864863ef52a0a237bc5805b8/integration_tests__tsan__[4_6].html)

```
>       raise Exception("Cannot wait RabbitMQ container")
E       Exception: Cannot wait RabbitMQ container
```

--- a/docker/test/integration/runner/misc/rabbitmq/enabled_plugins
+++ b/docker/test/integration/runner/misc/rabbitmq/enabled_plugins
@@ -1,0 +1,1 @@
+[rabbitmq_consistent_hash_exchange].

--- a/docker/test/integration/runner/misc/rabbitmq/rabbitmq.conf
+++ b/docker/test/integration/runner/misc/rabbitmq/rabbitmq.conf
@@ -13,3 +13,5 @@ ssl_options.fail_if_no_peer_cert = false
 ssl_options.cacertfile = /etc/rabbitmq/ca-cert.pem
 ssl_options.certfile = /etc/rabbitmq/server-cert.pem
 ssl_options.keyfile = /etc/rabbitmq/server-key.pem
+
+vm_memory_high_watermark.absolute = 2GB

--- a/tests/integration/compose/docker_compose_rabbitmq.yml
+++ b/tests/integration/compose/docker_compose_rabbitmq.yml
@@ -1,7 +1,9 @@
 services:
     rabbitmq1:
-        image: rabbitmq:3.12.6-alpine
+        image: rabbitmq:4.0.2-alpine
         hostname: rabbitmq1
+        environment:
+            RABBITMQ_FEATURE_FLAGS: feature_flags_v2,message_containers
         expose:
             - ${RABBITMQ_PORT:-5672}
             - ${RABBITMQ_SECURE_PORT:-5671}
@@ -14,3 +16,4 @@ services:
             - /misc/rabbitmq/ca-cert.pem:/etc/rabbitmq/ca-cert.pem
             - /misc/rabbitmq/server-cert.pem:/etc/rabbitmq/server-cert.pem
             - /misc/rabbitmq/server-key.pem:/etc/rabbitmq/server-key.pem
+            - /misc/rabbitmq/enabled_plugins:/etc/rabbitmq/enabled_plugins

--- a/tests/integration/helpers/cluster.py
+++ b/tests/integration/helpers/cluster.py
@@ -309,7 +309,8 @@ def check_rabbitmq_is_available(rabbitmq_id, cookie, timeout=90):
                 "await_startup",
             ),
             stderr=subprocess.STDOUT,
-            timeout=timeout)
+            timeout=timeout,
+        )
         return True
     except subprocess.CalledProcessError as e:
         # Raised if the command returns a non-zero exit code
@@ -320,8 +321,9 @@ def check_rabbitmq_is_available(rabbitmq_id, cookie, timeout=90):
         raise RuntimeError(error_message)
     except subprocess.TimeoutExpired as e:
         # Raised if the command times out
-        raise RuntimeError(f"RabbitMQ startup timed out. Output: {e.output.decode(errors='replace')}")
-
+        raise RuntimeError(
+            f"RabbitMQ startup timed out. Output: {e.output.decode(errors='replace')}"
+        )
 
 
 def rabbitmq_debuginfo(rabbitmq_id, cookie):

--- a/tests/integration/helpers/cluster.py
+++ b/tests/integration/helpers/cluster.py
@@ -298,19 +298,30 @@ def check_postgresql_java_client_is_available(postgresql_java_client_id):
     return p.returncode == 0
 
 
-def check_rabbitmq_is_available(rabbitmq_id, cookie):
-    p = subprocess.Popen(
-        docker_exec(
-            "-e",
-            f"RABBITMQ_ERLANG_COOKIE={cookie}",
-            rabbitmq_id,
-            "rabbitmqctl",
-            "await_startup",
-        ),
-        stdout=subprocess.PIPE,
-    )
-    p.wait(timeout=60)
-    return p.returncode == 0
+def check_rabbitmq_is_available(rabbitmq_id, cookie, timeout=90):
+    try:
+        subprocess.check_output(
+            docker_exec(
+                "-e",
+                f"RABBITMQ_ERLANG_COOKIE={cookie}",
+                rabbitmq_id,
+                "rabbitmqctl",
+                "await_startup",
+            ),
+            stderr=subprocess.STDOUT,
+            timeout=timeout)
+        return True
+    except subprocess.CalledProcessError as e:
+        # Raised if the command returns a non-zero exit code
+        error_message = (
+            f"RabbitMQ startup failed with return code {e.returncode}. "
+            f"Output: {e.output.decode(errors='replace')}"
+        )
+        raise RuntimeError(error_message)
+    except subprocess.TimeoutExpired as e:
+        # Raised if the command times out
+        raise RuntimeError(f"RabbitMQ startup timed out. Output: {e.output.decode(errors='replace')}")
+
 
 
 def rabbitmq_debuginfo(rabbitmq_id, cookie):
@@ -372,22 +383,6 @@ async def nats_connect_ssl(nats_port, user, password, ssl_ctx=None):
         tls=ssl_ctx,
     )
     return nc
-
-
-def enable_consistent_hash_plugin(rabbitmq_id, cookie):
-    p = subprocess.Popen(
-        docker_exec(
-            "-e",
-            f"RABBITMQ_ERLANG_COOKIE={cookie}",
-            rabbitmq_id,
-            "rabbitmq-plugins",
-            "enable",
-            "rabbitmq_consistent_hash_exchange",
-        ),
-        stdout=subprocess.PIPE,
-    )
-    p.communicate()
-    return p.returncode == 0
 
 
 def get_instances_dir(name):
@@ -2355,22 +2350,14 @@ class ClickHouseCluster:
         self.print_all_docker_pieces()
         self.rabbitmq_ip = self.get_instance_ip(self.rabbitmq_host)
 
-        start = time.time()
-        while time.time() - start < timeout:
-            try:
-                if check_rabbitmq_is_available(
-                    self.rabbitmq_docker_id, self.rabbitmq_cookie
-                ):
-                    logging.debug("RabbitMQ is available")
-                    if enable_consistent_hash_plugin(
-                        self.rabbitmq_docker_id, self.rabbitmq_cookie
-                    ):
-                        logging.debug("RabbitMQ consistent hash plugin is available")
-                    return True
-                time.sleep(0.5)
-            except Exception as ex:
-                logging.debug("Can't connect to RabbitMQ " + str(ex))
-                time.sleep(0.5)
+        try:
+            if check_rabbitmq_is_available(
+                self.rabbitmq_docker_id, self.rabbitmq_cookie, timeout
+            ):
+                logging.debug("RabbitMQ is available")
+                return True
+        except Exception as ex:
+            logging.debug("RabbitMQ await_startup failed", exc_info=True)
 
         try:
             with open(os.path.join(self.rabbitmq_dir, "docker.log"), "w+") as f:

